/*
  CarStandFCOOL API

  An API for managing a cloud-based car dealership service

  The version of the OpenAPI document: 1.0.0

  Generated by OpenAPI Generator: https://openapi-generator.tech
*/

syntax = "proto3";

package openapitools.services.carservice;

import "google/protobuf/empty.proto";
import public "models/car.proto";

service CarService {
  rpc CarsCreate (CarsCreateRequest) returns (Car);

  rpc CarsDelete (CarsDeleteRequest) returns (google.protobuf.Empty);

  rpc CarsReadAll (google.protobuf.Empty) returns (CarsReadAllResponse);

  rpc CarsReadOne (CarsReadOneRequest) returns (Car);

  rpc CarsUpdate (CarsUpdateRequest) returns (Car);

}

message CarsCreateRequest {
  // Car to add
  Car car = 1;

}

message CarsDeleteRequest {
  // ID of the car
  int32 carId = 1;

}

message CarsReadAllResponse {
  repeated Car data = 1;
}

message CarsReadOneRequest {
  // ID of the car
  int32 carId = 1;

}

message CarsUpdateRequest {
  // ID of the car
  int32 carId = 1;
  Car car = 2;

}

